components/Common/Select/index.tsx

```diff
@@ -1,12 +1,13 @@
 import { ChevronDownIcon } from '@heroicons/react/24/outline';
 import * as Primitive from '@radix-ui/react-select';
+import Image from 'next/image';
 import { useId } from 'react';
 import type { FC } from 'react';
 
 import styles from './index.module.css';
 
 type SelectProps = {
-  values: ({ label: string; value: string } | string)[];
+  values: ({ label: string; value: string; iconImageUrl?: string } | string)[];
```

Comment:
Can you extract `{ label: string; value: string; iconImageUrl?: string }` into a separate type? It can be in the same file.

Reply:
Also this type is wrong as it allows the same array to have plain strings and objects. A correct type would be:

```tsx
values: { label: string; value: string; iconImageUrl?: string }[] | string[]
```

Reply:
And the reason of why either supporting one or another, is for the sake of consistency and to reduce chances of having things breaking apart because some entries are strings and others are objects

Reply:
thank you for the explanation 👍